core/new-gui/src/app/dashboard/component/feature-container/user-project-section/project-page/project-page.component.ts

```diff
@@ -0,0 +1,306 @@
+        if (confirmToDelete && wid !== undefined) {
+          this.workflowPersistService
+            .deleteWorkflow(wid)
+            .pipe(untilDestroyed(this))
+            .subscribe(
+              _ => {
+                this.workflows = this.workflows.filter(
+                  workflowEntry => workflowEntry.workflow.wid !== wid
+                );
+              },
+              // @ts-ignore // TODO: fix this with notification component
+              (err: unknown) => alert(err.error)
+            );
+        }
+      });
+  }
+
+
+
+  // ----------------- for file card
+  public addFileSizeUnit(fileSize: number): string {
+    return this.userFileService.addFileSizeUnit(fileSize);
+  }
+
+  public confirmEditFileName(
+    dashboardUserFileEntry: DashboardUserFileEntry,
+    name: string,
+    index: number
+  ): void {
+    const {
+      file: { fid },
+    } = dashboardUserFileEntry;
+    this.userFileService
+      .updateFileName(fid, name)
+      .pipe(untilDestroyed(this))
+      .subscribe(
+        () => {
+          this.userFileService.refreshDashboardUserFileEntries();
+          this.userProjectService.refreshFilesOfProject(this.pid); // -- perform appropriate call for project page
+        },
+        (err: unknown) => {
+          // @ts-ignore // TODO: fix this with notification component
+          this.notificationService.error(err.error.message);
+          this.userFileService.refreshDashboardUserFileEntries();
+          this.userProjectService.refreshFilesOfProject(this.pid); // -- perform appropriate call for project page
+        }
+      )
+      .add(() => (this.isEditingFileName = this.isEditingFileName.filter(fileIsEditing => fileIsEditing != index)));
+  }
+
+  public onClickOpenFileShareAccess(dashboardUserFileEntry: DashboardUserFileEntry): void {
+    const modalRef = this.modalService.open(NgbdModalUserFileShareAccessComponent);
+    modalRef.componentInstance.dashboardUserFileEntry = dashboardUserFileEntry;
+  }
+
+  public downloadUserFile(userFileEntry: DashboardUserFileEntry): void {
+    this.userFileService
+      .downloadUserFile(userFileEntry.file)
+      .pipe(untilDestroyed(this))
+      .subscribe(
+        (response: Blob) => {
+          // prepare the data to be downloaded.
+          const dataType = response.type;
+          const binaryData = [];
+          binaryData.push(response);
+
+          // create a download link and trigger it.
+          const downloadLink = document.createElement("a");
+          downloadLink.href = URL.createObjectURL(new Blob(binaryData, { type: dataType }));
+          downloadLink.setAttribute("download", userFileEntry.file.name);
+          document.body.appendChild(downloadLink);
+          downloadLink.click();
+          URL.revokeObjectURL(downloadLink.href);
+        },
+        (err: unknown) => {
+          // @ts-ignore // TODO: fix this with notification component
+          this.message.error(err.error.message);
```

Comment:
Can we use notification service here? e.g. `this.notificationService.error(err.error.message);`

Reply:
please use notification service.

Reply:
done